src/adk/api/local_api.py

```diff
@@ -525,6 +533,75 @@ def __is_structure_valid(self, application_path: Path, error_dict: ErrorDictType
         if not (application_path / 'MANIFEST.ini').is_file():
             error_dict["warning"].append(f"{application_path} should contain the file 'MANIFEST.ini'")
 
+    def __is_python_valid(self, application_path: Path, error_dict: ErrorDictType) -> None:
+        """
+        Validate if the python code in app_{role}.py has valid syntax
+
+        Args:
+            application_path: Path of where the application is located
+            error_dict: A dictionary for storing the validation errors
+
+        """
+        app_config_path = application_path / 'config'
+        app_src_path = application_path / 'src'
+
+        role_file_names = self.__get_role_file_names(app_config_path)
+        for role_file in role_file_names:
+            if (app_src_path / role_file).is_file():
+                valid, validation_message = utils.check_python_syntax(app_src_path / role_file)
+                if not valid:
+                    error_dict["error"].append(f"The application source file {role_file} has syntax errors"
+                                               f"\n {validation_message}")
+                else:
+                    # The syntax is valid. So we can now check for the input params to the main()
+                    self.__is_valid_input_params_for_role(application_path, role_file, error_dict)
+
+    def __is_valid_input_params_for_role(self, application_path: Path, role_file: str, error_dict: ErrorDictType) \
+        -> None:
+        """
+        Validate if the inputs defined in application config are passed as parameters to main() in app_{role}.py
+
+        Args:
+            application_path: Path of where the application is located
+            role_file: Name of the file in which function is defined
+            error_dict: A dictionary for storing the validation errors
+
+        """
+        role_input_mapping = self.__get_role_input_mapping(application_path)
+        role_name = self.__get_role_name_from_role_file(role_file)
+
+        if role_name:
+            app_src_path = application_path / 'src'
+            if (app_src_path / role_file).is_file():
+                arg_list = utils.get_function_arguments(app_src_path / role_file, function_name='main')
```

Comment:
You won't have to give the default value for function_name (I think best is remove the default value in this method in utils).

Reply:
Fixed